### PR TITLE
Kafka Sample Improvements

### PR DESCRIPTION
**Pull request by garyrussell**
- get rid of topic creator and scala deps
- use boot properties
- demonstrate using the DSL to dynamically add adapters

When reviewed/approved, I can make similar changes to the DSL version.

### CODE REVIEW

File: basic/kafka/README.md [hunk 1]
```diff
@@ -19,21 +19,36 @@ In STS (Eclipse), go to package **org.springframework.integration.samples.kafka*
 
 ### Output
 
-The application sends 10 messages (`foo0` ... `foo9`) to a kafka topic `si.topic` (which is created if necessary).
+The application sends 10 messages (`foo0` ... `foo9`) to a kafka topic `si.topic`; this must exist or the broker must be configured to auto-create topics.
```
Comment (artembilan) [inline basic/kafka/README.md]: ??? You just don't want to promote our "admin utils" in test module. Do you ?

File: basic/kafka/src/main/java/org/springframework/integration/samples/kafka/Application.java [hunk 2]
```diff
@@ -51,44 +55,55 @@
...
 	public static void main(String[] args) throws Exception {
 		ConfigurableApplicationContext context
 				= new SpringApplicationBuilder(Application.class)
-				.web(false)
-				.run(args);
+					.web(WebApplicationType.NONE)
+					.run(args);
+		context.getBean(Application.class).runDemo(context);
+		context.close();
+		System.exit(0);
```
Comment (artembilan) [inline basic/kafka/src/main/java/org/springframework/integration/samples/kafka/Application.java]: Do we really need `System.exit()` in the end of `main()`?
Especially when we `close()` ctx, therefore stop all the async processes.

File: build.gradle [hunk 3]
```diff
@@ -198,7 +198,7 @@ subprojects { subproject ->
 			postgresVersion = '9.1-901-1.jdbc4'
 			subethasmtpVersion = '1.2'
 			slf4jVersion = '1.7.11'
-			springIntegrationVersion = '5.0.0.M1'
+			springIntegrationVersion = '5.0.0.BUILD-SNAPSHOT'
 			springIntegrationKafkaVersion = '3.0.0.BUILD-SNAPSHOT'
 			springIntegrationSplunkVersion = '1.1.0.RELEASE'
 			springKafkaVersion = '1.1.2.BUILD-SNAPSHOT'
```
Comment (artembilan) [inline build.gradle]: Shouldn't we move to `SK-2.0` already ?

File: basic/kafka/src/main/java/org/springframework/integration/samples/kafka/Application.java (hunk 2 above)
Comment (garyrussell) [inline basic/kafka/src/main/java/org/springframework/integration/samples/kafka/Application.java]: I don't remember why it's there but I just tested without it and there are no problems.

File: build.gradle (hunk 3 above)
Comment (garyrussell) [inline build.gradle]: 👍 

Can you do these minor polishes on merge?

We can consider re-adding topic provisioning when we create the admin module.

**Comment by artembilan** [comment]
Merged as https://github.com/spring-projects/spring-integration-samples/commit/038ee4576ac2119958e5c8d4ad2681402cd60e2e

**Pull request closed without merging**